grpc/src/test/java/com/linecorp/armeria/server/grpc/UnframedGrpcErrorHandlerBuilderTest.java

```diff
@@ -0,0 +1,197 @@
+                                              .jsonMarshaller(
+                                                      UnframedGrpcErrorHandlers.ERROR_DETAILS_MARSHALLER))
+                .isInstanceOf(IllegalStateException.class)
+                .hasMessageContaining(
+                        "Cannot set a custom JSON marshaller because one or more Message instances or " +
+                        "Message types have already been registered. To set a custom marshaller, " +
+                        "ensure that no Message or Message type registrations have been made before " +
+                        "calling this method.");
+    }
+
+    @Test
+    void buildWithoutOptions() {
+        final UnframedGrpcErrorHandler unframedGrpcErrorHandler = UnframedGrpcErrorHandler.builder().build();
+        final ServiceRequestContext ctx =
+                ServiceRequestContext.of(HttpRequest.of(RequestHeaders.of(HttpMethod.GET, "/test")));
+        final AggregatedHttpResponse jsonResponse =
+                AggregatedHttpResponse.of(HttpStatus.INTERNAL_SERVER_ERROR,
+                                          MediaType.JSON_UTF_8,
+                                          "{\"message\":\"Internal Server Error\"}");
+        final HttpResponse httpResponseJson =
+                unframedGrpcErrorHandler.handle(ctx, Status.INTERNAL, jsonResponse);
+        final AggregatedHttpResponse aggregatedHttpResponse = httpResponseJson.aggregate().join();
+        assertThat(aggregatedHttpResponse.headers().contentType().isJson()).isTrue();
+
+        final AggregatedHttpResponse plaintextResponse =
+                AggregatedHttpResponse.of(HttpStatus.INTERNAL_SERVER_ERROR,
+                                          MediaType.PLAIN_TEXT_UTF_8,
+                                          "Internal Server Error");
+        final HttpResponse httpResponsePlaintext =
+                unframedGrpcErrorHandler.handle(ctx, Status.INTERNAL, plaintextResponse);
+        assertThat(httpResponsePlaintext.aggregate().join().headers()
+                                        .contentType().is(MediaType.PLAIN_TEXT)).isTrue();
+    }
+
+    @Test
+    void buildWithResponseType() {
+        final UnframedGrpcErrorHandler unframedGrpcErrorHandlerJson =
+                UnframedGrpcErrorHandler.builder()
+                                        .responseTypes(UnframedGrpcErrorResponseType.JSON)
+                                        .build();
+        final ServiceRequestContext ctx = ServiceRequestContext.of(
+                HttpRequest.of(RequestHeaders.of(HttpMethod.GET, "/test")));
+        final AggregatedHttpResponse response = AggregatedHttpResponse.of(HttpStatus.INTERNAL_SERVER_ERROR);
+        final HttpResponse httpResponseJson =
+                unframedGrpcErrorHandlerJson.handle(ctx, Status.INTERNAL, response);
+        assertThat(requireNonNull(httpResponseJson.aggregate().join().headers().contentType())
```

Comment:
You need to remove `requireNonNull()` from all `assertThat()` calls in this PR as well :wink:

Reply:
Sorry! I thought I removed all.